jcl/src/java.base/share/classes/java/lang/StackWalker.java

```diff
@@ -376,6 +402,33 @@ public StackTraceElement toStackTraceElement() {
 					lineNumber);
 		}
 
+		/*[IF Java10]*/
+		/**
+		 * Creates a MethodType object for the method associated with this frame.
+		 * @throws UnsupportedOperationException if the StackWalker object is not configured with RETAIN_CLASS_REFERENCE
+		 * @return MethodType object
+		 * @since 10
+		 */
+		@Override
+		public MethodType getMethodType() {
+			if (null == declaringClass) {
+				/*[MSG "K0639","Stack walker not configured with RETAIN_CLASS_REFERENCE"]*/
+				throw new UnsupportedOperationException(com.ibm.oti.util.Msg.getString("K0639")); //$NON-NLS-1$
+			}
+			return MethodType.fromMethodDescriptorString(methodSignature, declaringClass.getClassLoader());
```

Comment:
Should `getClassLoader()` be wrapped in a `doPrivileged()` block?  Or use the package private `getClassLoaderImpl()` to avoid the security check here.

Reply:
> Should getClassLoader() be wrapped in a doPrivileged() block?
Not necessary as getClassLoader() checks the caller loader before doing any security check.

> Or use the package private getClassLoaderImpl() to avoid the security check here.
getClassLoaderImpl() doesn't return the same result as getClassLoader() we should add a new helper that works the same and avoid the stack walk.

Reply:
I added Class.internalGetClassLoader() which is basically getClassLoader() w/o security checks.
Please review and I will amend the changes, then re-test.